compiler/src/iree/compiler/Dialect/HAL/Target/LLVM/LLVMCPUTarget.cpp

```diff
@@ -410,13 +412,38 @@ class LLVMCPUTargetBackend final : public TargetBackend {
         continue;
       }
       func.setDSOLocal(true);
-      func.setLinkage(llvm::GlobalValue::LinkageTypes::InternalLinkage);
+      func.setVisibility(llvm::GlobalValue::VisibilityTypes::DefaultVisibility);
+      func.setLinkage(llvm::GlobalValue::LinkageTypes::PrivateLinkage);
+      func.setDLLStorageClass(
+          llvm::GlobalValue::DLLStorageClassTypes::DefaultStorageClass);
     }
     for (auto &global : llvmModule->getGlobalList()) {
       global.setDSOLocal(true);
       global.setLinkage(llvm::GlobalValue::LinkageTypes::InternalLinkage);
     }
 
+    // Since all exports are fetched via the executable library query function
+    // we don't need to make them publicly exported on the module. This has
+    // the downside of making some tooling (disassemblers/decompilers/binary
+    // size analysis tools/etc) a bit harder to work with, though, so when
+    // compiling in debug mode we export all the functions.
+    if (options_.debugSymbols) {
```

Comment:
I wonder if we should be turning this off for benchmarks or call it out somewhere in user docs...

https://github.com/iree-org/iree/blob/56a873934c70a440e26f39ae14eb82ba0ed54897/compiler/src/iree/compiler/Dialect/HAL/Target/LLVM/LLVMTargetOptions.h#L43-L47 https://github.com/iree-org/iree/blob/56a873934c70a440e26f39ae14eb82ba0ed54897/compiler/src/iree/compiler/Dialect/HAL/Target/LLVM/LLVMTargetOptions.cpp#L155-L159

Reply:
Should we store debug symbols off by default, and enable them in IREEs debug builds?

Reply:
We'd want the opposite - this should be _on_ for all benchmarks - nothing here impacts runtime performance (or shouldn't), just provides more information for the kind of tools that'd be used in benchmarking.

Reply:
Ah, okay. Default to enabling debug symbols, only disable debug symbols when trying to minimize binary size? Does the "significantly change the output program and benchmarking" comment only apply when setting debug symbols separately from opt level?